README.md

```diff
@@ -132,7 +132,55 @@ Task lock is implemented to prevent task cache collision.
     redis_port=6379
     ```
 
-### Inherit task parameters with decorator
+### 2. Using efficient task cache collision lock
+#### Description
+Above task lock will prevent cache collision. However, above setting check collisions only when the task access the cache file (i.e. `task.dump()`, `task.load()` and `task.remove()`). This will allow applications to run `run()` of same task at the same time, which is not efficent.
+
+Settings in this section will prevent running `run()` at the same time for efficiency.
+
+1. Set normal cache collision lock
+    Set cache collision lock following `1. Task cache collision lock`.
+
+2. Decorate `run()` with `@RunWithLock`
+    Decorate `run()` of yourt gokart tasks which you want to lock with `@RunWithLock`.
+
+    ```python
+    from gokart.run_with_lock import RunWithLock
+    
+    @RunWithLock
+    class SomeTask(gokart.TaskOnKart):
+        def run(self):
+            ...
+    ```
+
+3. Set `redis_fail` parameter to true.
```

Comment:
@mski-iksm 
`redis_fail` is not comprehensive to me. How about `fail_on_collision` `fail_on_redis_collision` or `wait_with_lock` (inverted).

Reply:
@Hi-king U've changed to redis_fail_on_collision !